### ISSUE TITLE
Invalid Encryption Key parameters with multiple Artifact Stores in CodePipeline

### ISSUE THREAD

**Issue by bensechrist** [awslabs/aws-cdk#1892]
When using a custom encryption key for SSE on the primary artifact bucket on a cross-region pipeline, the YAML produced for the ArtifactStore is invalid. See code example:

```ts
import { CloudFormationCapabilities, PipelineCreateReplaceChangeSetAction } from '@aws-cdk/aws-cloudformation';
import codepipeline = require('@aws-cdk/aws-codepipeline');
import { Artifact, ArtifactPath } from '@aws-cdk/aws-codepipeline-api';
import iam = require('@aws-cdk/aws-iam');
import cdk = require('@aws-cdk/cdk');
import { Secret } from '@aws-cdk/cdk';
import { Bucket, BucketEncryption } from '@aws-cdk/aws-s3';
import { EncryptionKey } from '@aws-cdk/aws-cloudformation/node_modules/@aws-cdk/aws-kms';

export class TestStack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
    super(scope, id, { env: { account: '13255', region: 'eu-west-1' }, ...props});

    const crossAccountId = '0987654321111';
    const stage = 'prod';
    const stackName = 'Henk';
    const changeSetName = 'Biertje';

    const artifactBucket = new Bucket(this, 'ArtifactBucket', {
      bucketName: 'test-bucket',
      encryptionKey: new EncryptionKey(this, 'CrossAccountKmsKey'),
      encryption: BucketEncryption.Kms,
    });
    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
      artifactBucket,
    });

    const sourceStage = new codepipeline.Stage(this, 'SourceStage', {
      pipeline
    });
    const action = new codepipeline.GitHubSourceAction(this, 'Action', {
      owner: 'bier',
      repo: 'lekker',
      stage: sourceStage,
      oauthToken: new Secret()
    });

    const region = 'us-east-1'; // Current stack and Pipeline are in us-west-2
    const crossAccountRoleArn = `arn:aws:iam::${crossAccountId}:role/CrossAccountRole`;
    const cloudformationRoleArn = `arn:aws:iam::${crossAccountId}:role/CloudformationRole`;
    const crossAccountRole: iam.IRole = iam.Role.import(this, 'CrossAccountRole', {
      roleArn: crossAccountRoleArn
    });
    const cloudformationRole: iam.IRole = iam.Role.import(this, 'CloudformationRole', {
      roleArn: cloudformationRoleArn
    });

    const stageSpecificStage = new codepipeline.Stage(this, 'Stage', {
      pipeline
    });
    new PipelineCreateReplaceChangeSetAction(this, `MakeChangeSet${stage}-${region}`, {
      region,
      stackName,
      changeSetName,
      stage: stageSpecificStage,
      runOrder: 1,
      adminPermissions: false,
      role: crossAccountRole,
      deploymentRole: cloudformationRole,
      templatePath: new ArtifactPath(new Artifact(action, 'henk'), 'asdf'),
      capabilities: CloudFormationCapabilities.AnonymousIAM,
    });
  }
}
```

The YAML produced uses lowercase `type` and `id` for the EncryptionKey parameters of the main artifact bucket

```yaml
      ArtifactStores:
        - Region: us-east-1
          ArtifactStore:
            Location: cdk-cross-region-codepipeline-replication-bucket-fd0f4cb887c4
            Type: S3
        - Region: us-west-2
          ArtifactStore:
            Location:
              Ref: ArtifactBucket7410C9EF
            Type: S3
            EncryptionKey:
              type: KMS
              id:
                Fn::GetAtt:
                  - CrossAccountKmsKeyFAA2FC05
                  - Arn
```

**Comment by skinny85** [comment]
Thanks for reporting @bensechrist !

**Comment by bensechrist** [comment]
Any update on this?

**Comment by skinny85** [comment]
This is still on our radar, but we haven't had time to tackle this one yet, sorry.

**Comment by skinny85** [comment]
I'm working on this issue right now.

Interestingly, with the current version (`0.34.0`), I'm not able to reproduce this bug.

Can someone see if they can post a minimal reproduction with `0.34.0`?

Thanks,
Adam

**Comment by skinny85** [comment]
I've done extensive unit & manual tests, and I'm 99% certain this bug has been fixed from version `0.34.0` onwards. I'm resolving this one, if anybody encounters this issue in version `0.34.0` or later, please comment.

Thanks,
Adam
